src/DependencyInjection/Configuration.php

```diff
@@ -24,6 +24,148 @@
  * sections are normalized, and merged.
  *
  * @author Michael Williams <mtotheikle@gmail.com>
+ *
+ * @phpstan-import-type ExtensionMap from \Sonata\AdminBundle\DependencyInjection\Compiler\ExtensionCompilerPass
+ * @phpstan-import-type Item from \Sonata\AdminBundle\Admin\Pool
+ *
+ * @phpstan-type SonataAdminConfigurationOptions = array{
+ *     confirm_exit: bool,
+ *     default_admin_route: string,
+ *     default_group: string,
+ *     default_icon: string,
+ *     default_label_catalogue: string,
+ *     dropdown_number_groups_per_colums: int,
+ *     form_type: 'standard'|'horizontal',
+ *     html5_validate: bool,
+ *     js_debug: bool,
+ *     list_action_button_content: 'text'|'icon'|'all',
+ *     lock_protection: bool,
+ *     logo_content: 'text'|'icon'|'all',
+ *     mosaic_background: string,
+ *     pager_links: int|null,
+ *     skin: 'skin-black'|'skin-black-light'|'skin-blue'|'skin-blue-light'|'skin-green'|'skin-green-light'|'skin-purple'|'skin-purple-light'|'skin-red'|'skin-red-light'|'skin-yellow'|'skin-yellow-light',
+ *     sort_admins: bool,
+ *     use_bootlint: bool,
+ *     use_icheck: bool,
+ *     use_select2: bool,
+ *     use_stickyforms: bool,
+ * }
+ *
+ * @phpstan-type SonataAdminConfiguration = array{
```

Comment:
This structure looks very complex. Maybe it's better to create a phpstan extension that can guess this structure based on the `Configuration` class.

Reply:
To me, phpstan extensions are made to understand code, not generating some phpdoc.
And I need this type to have a correct phpdoc in tests or in some other classes.

Reply:
To me it only reflects the complexity of the Sonata Admin Bundle configuration options. What should be correctly specified is the opcional keys. All this is always generated? I mean, if you use Sonata with the minimal configuration needed, still all the array is generated or some of it is omited?

Reply:
> All this is always generated? I mean, if you use Sonata with the minimal configuration needed, still all the array is generated or some of it is omited?

I need to check again but this is what I tried to defined, in order to write
```
/** @phpstan-var SonataAdminConfiguration $config */
$config = (new Processor())->processConfiguration(new Configuration(), []);
```